packages/opentelemetry-sdk-trace-base/test/common/export/BatchSpanProcessorBase.test.ts

```diff
@@ -222,14 +214,14 @@ describe('BatchSpanProcessorBase', () => {
 
     it('should force flush on demand', () => {
       const processor = new BatchSpanProcessor(exporter, defaultBufferConfig);
-      for (let i = 0; i < defaultBufferConfig.maxExportBatchSize; i++) {
-        const span = createSampledSpan(`${name}_${i}`);
+      const span = createSampledSpan(name);
+      for (let i = 1; i < defaultBufferConfig.maxExportBatchSize; i++) {
         processor.onStart(span, ROOT_CONTEXT);
         processor.onEnd(span);
       }
       assert.strictEqual(exporter.getFinishedSpans().length, 0);
       processor.forceFlush();
-      assert.strictEqual(exporter.getFinishedSpans().length, 5);
+      assert.strictEqual(exporter.getFinishedSpans().length, 4);
```

Comment:
Same question here

Reply:
Same answer, we want to avoid the eager behaviour and as such emit one fewer spans than before